crypto/sha/asm/keccak1600-armv4.pl

```diff
@@ -966,6 +966,8 @@ sub Round {
 	stmdb	sp!,{r6-r9}
 
 	mov	r14,$A_flat
+	cmp	r4, #0                 @ r4 = 'next' argument
```

Comment:
r4 is clobbered by line 945 because $out is r4.

Reply:
Bigger problem was that first 4 parameters are passed using r0..r3, the rest are on the stack. Also had trouble finding a free register, ended up using r11.